pkg/audit/verifier_test.go

```diff
@@ -514,3 +515,59 @@ func TestVerifierModifiedSegment(t *testing.T) {
 		assert.Empty(t, report)
 	})
 }
+
+func TestVerifierModifiedSegmentFailsOnce(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		audits := planet.Satellites[0].Audit.Service
+		err := audits.Close()
+		require.NoError(t, err)
+
+		ul := planet.Uplinks[0]
+		testData := testrand.Bytes(8 * memory.KiB)
+
+		err = ul.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		stripe, _, err := audits.Cursor.NextStripe(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, stripe)
+
+		// delete the piece from the first node
+		origNumPieces := len(stripe.Segment.GetRemote().GetRemotePieces())
+		piece := stripe.Segment.GetRemote().GetRemotePieces()[0]
+		pieceID := stripe.Segment.GetRemote().RootPieceId.Derive(piece.NodeId, piece.PieceNum)
+		node := getStorageNode(planet, piece.NodeId)
+		err = node.Storage2.Store.Delete(ctx, planet.Satellites[0].ID(), pieceID)
+		require.NoError(t, err)
+
+		report, err := audits.Verifier.Verify(ctx, stripe, nil)
+		require.NoError(t, err)
+
+		assert.Len(t, report.Successes, origNumPieces-1)
+		assert.Len(t, report.Fails, 1)
+		assert.Equal(t, report.Fails[0], piece.NodeId)
+		assert.Len(t, report.Offlines, 0)
+		require.Len(t, report.PendingAudits, 0)
+
+		//refetch the stripe
+		stripe, _, err = audits.Cursor.NextStripe(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, stripe)
+
+		report, err = audits.Verifier.Verify(ctx, stripe, nil)
+		require.NoError(t, err)
+
+		//verify no failures because that segment is gone
+		assert.Len(t, report.Successes, origNumPieces-1)
+		assert.Len(t, report.Fails, 0)
+		assert.Len(t, report.Offlines, 0)
+		require.Len(t, report.PendingAudits, 0)
+
+		for _, newPiece := range stripe.Segment.GetRemote().GetRemotePieces() {
+			require.NotEqual(t, newPiece.NodeId, piece.NodeId)
```

Comment:
This should be `assert` too. We want the test case to report all mismatches.

Reply:
I'm not sure if matters, but fixing

Reply:
Imagine what a massive bug we would find if this check fails more than once.